### ISSUE TITLE
Persist/cache processed crossfilter object

### ISSUE THREAD

**Issue by rmitsch** [crossfilter/crossfilter#110]
Is there a possibility to dump or cache a crossfilter object? 

My use case is the following: I have a large-ish dataset which takes some time (tens of seconds) to process with crossfilter, since I have a non-trivial number of dimensions. Hence I would like to cache my crossfilter.js object between page reloads to shorten waiting times, but serialization/putting it into localStorage renders the object unusable, i. e. trying to use it yields errors. I tried various options, including https://github.com/hunterloftis/cryo, none of which produced the desired result - a (serialized) object that can be de-serialized and converted into a working crossfilter object.

Therefore I was wondering whether there is an option to somehow persist or cache the crossfilter object (not the data I feed into crossfilter!). I'm aware of the option of pre-aggregating my data further, but I'm looking to avoid this approach.

I appreciate your time & thanks for this great library!

**Comment by gordonwoodhull** [comment]
In principal, all of the internal data *could* be serialized, e.g. the crossfilter object data

https://github.com/crossfilter/crossfilter/blob/7d8d58a895927206a88fa8caf8e11a4226ef999a/src/crossfilter.js#L44-L46

the dimension data (there are a lot more here, this is just for example:

https://github.com/crossfilter/crossfilter/blob/7d8d58a895927206a88fa8caf8e11a4226ef999a/src/crossfilter.js#L161-L166

and the group data (also probably more here):

https://github.com/crossfilter/crossfilter/blob/7d8d58a895927206a88fa8caf8e11a4226ef999a/src/crossfilter.js#L776-L780

What would be more difficult is that in general, functions can not be serialized. So certainly none of the callbacks/listeners could be serialized, and the key/group/reduce functions could only be serialized if they use nothing from their environment/closures. I don't think there's any way to test this, and it's really very common that those functions do use external variables.

I haven't done a lot of serialization of functions, but I understand that as long as the function does not use a closure, you can use `.toString()` and `eval` to serialize/deserialize, but IIRC this is only reliable post ES6.

In short, it's a significant engineering effort, but if you want to give it a shot I think the community should be able to help you out. I know it will get use if it works.

With reference to [your original question](https://stackoverflow.com/questions/52027944/javascript-temporary-storage-of-large-10-mb-object-data?noredirect=1#comment91592545_52027944), it seems like overkill to me. I guess it depends how much time and energy you have on your hands. 😉
